packages/SwingSet/src/kernel/kernel.js

```diff
@@ -821,8 +823,13 @@ export default function buildKernel(
     }
     const { meterID } = vatInfo;
     const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
+    const disconnectObject = {
+      name: 'vatUpgraded',
+      upgradeMessage,
+      seqNum: vatKeeper.getSeqNum(),
+    };
     /** @type { import('../types-external.js').KernelDeliveryStopVat } */
-    const kd1 = harden(['stopVat']);
+    const kd1 = harden(['stopVat', disconnectObject]);
```

Comment:
In general, `VatDeliveryObject`s like this `kd1` are populated with capdata (i.e. `dispatch.deliver` or the individual promise resolutions of `dispatch.notify`). I'm a little on the fence about it, but if we maintain that practice, then we should do:
```js
const disconnectObjectCD = harden({ body: JSON.stringify(disconnectObject), slots: []});
const kd1 = harden(['stopVat', disconnectObjectCD]);
```

and either change liveslots to do a `disconnectObject = m.unserialize(disconnectObjectCD)` at the top of `stopVat(disconnectObjectCD)`, or remove the `m.serialize(disconnectObject)` from `rejectAllPromises()`. Also the type definition in types-external.js would need to change.

The vatTranslator.js code that translates `stopVat` could mostly stay the same, but it would want an `insistCapData` check and also an `assert.equal(disconnectObjectCD.slots.length, 0)` to guard against us allowing slots in the future and forgetting to add `convertKernelSlotToVatSlot` calls to the translator (which would leak untranslated krefs into the vat).

On one hand, consistency is nice, and this paves the way for more complicated `disconnectObject` contents. It would take more work on both sides to allow krefs/vrefs in the object (`vatTranslator.js` would need updating, but also liveslots has more bookkeeping to do if/when it sends slots into the kernel), but we'd be halfway there.

On the other hand, I don't want to allow slots in a `disconnectObject` any time soon, so most of this would feel like self-inflicted busywork.

The one nice thing about using capdata is that changes to the contents of the disconnect object would not be visible to the liveslots *types*: if we added a fourth property, or changed the name of the `seqNum` parameter, the `VatDeliveryObject` shape would remain exactly the same. In practice it wouldn't matter: liveslots just serializes whatever you give it. But in principle, if we change `DisconnectObject` within the kernel, we've just changed the API signature of the liveslots `stopVat`, and that's running in a different process. If the rule is "it's just capdata", then they're the same.

Not sure what to recommend.

Reply:
The necessary change to `watchedPromiseManager.prepareShutdownRejections` pushes me slightly in favor of going with capdata, because now we're introducing a second API boundary: this one between the current version of liveslots and some future version of liveslots (communicating through the vatstore DB). "Uniform capdata with empty `.slots`" is a decent type to share across versions.

Reply:
One benefit (albeit principally just an aesthetic benefit) would be to make `startVat` and `stopVat` have parallel form.